mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/model/MailboxMessage.java

```diff
@@ -34,6 +34,8 @@
  */
 public interface MailboxMessage extends Message, Comparable<MailboxMessage> {
 
+    void setThreadId(ThreadId threadId);
```

Comment:
No I would definitly prefer to have threadId final and add the required onstructor parameters!